### ISSUE TITLE
Option to Automatically Infer if CSV file has a header

### ISSUE THREAD

**Issue by domoritz** [apache/arrow-rs#6882]
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**

When parsing CSV files, I don't want to always provide information about e.g. the header (which defaults to false). It would be nice if the format could be inferred like the schema. 

**Describe the solution you'd like**

Add an `infer_format` method.

**Comment by tustvold** [comment]
https://docs.rs/arrow-csv/latest/arrow_csv/reader/struct.Format.html#method.infer_schema

**Comment by domoritz** [comment]
Yes, I am looking after inferring the format, not the schema. I am especially interested in the header since the default is false but many CSV files have a header.

**Comment by tustvold** [comment]
Unfortunately I can't think of a reliable way to do this

**Comment by domoritz** [comment]
There is not perfect way but if you find that the first row has strings while the subsequent ones have e.g. numbers it's probably a header. 

Iirc duckdb had a pretty good automatic inference algorithm for csv parsing.

**Comment by Jo2234** [comment]
take

**Comment by Rich-T-kid** [comment]
related 
- [ ] https://github.com/apache/arrow-rs/issues/2580

**Comment by Jo2234** [comment]
This issue was automatically closed when #11105 merged, but that PR only fixes leading-plus numeric schema inference. Header detection is still pending in #10361. I corrected the accidental closing-keyword wording in my PR description. Could a maintainer please reopen this issue? Sorry for the bookkeeping error.
